cosmic-settings/src/pages/power/backend/mod.rs

```diff
@@ -270,81 +315,149 @@ async fn get_on_battery_status() -> Result<bool, zbus::Error> {
+
+        let icon_name =
+            format!("cosmic-applet-battery-level-{battery_percent}-{charging}symbolic",);
+
+        let remaining_time = |duration: Duration| {
+            let total_seconds = duration.num_seconds();
+
+            let hours = total_seconds / 3600;
+            let minutes = (total_seconds % 3600) / 60;
+            let seconds = total_seconds % 60;
+
+            fl!(
+                "battery",
+                "remaining-time",
+                time = format!("{:02}:{:02}:{:02}", hours, minutes, seconds)
+            )
+        };
+
+        Self {
+            icon_name,
+            is_present,
+            percent,
+            on_battery,
+            remaining_duration,
+            remaining_time: remaining_time(remaining_duration),
+        }
+    }
+
+    pub async fn update_battery() -> Self {
+        let proxy = get_device_proxy().await;
+
+        if let Ok(proxy) = proxy {
+            return Self::from_device(proxy).await;
         }
 
         Battery::default()
     }
 }
+
+impl ConnectedDevice {
+    async fn from_device_maybe(proxy: DeviceProxy<'_>) -> Option<Self> {
+        let device_type = proxy.type_().await.unwrap_or(BatteryType::Unknown);
+        if matches!(
+            device_type,
+            BatteryType::Unknown | BatteryType::LinePower | BatteryType::Battery
+        ) {
+            return None;
+        }
+        let model = proxy
+            .model()
+            .await
+            .unwrap_or(fl!("connected-devices", "unknown"));
+        let battery = Battery::from_device(proxy).await;
+        let device_icon = match device_type {
+            BatteryType::Ups => "uninterruptible-power-supply-symbolic",
+            BatteryType::Monitor => "display-symbolic",
+            BatteryType::Mouse => "input-mouse-symbolic",
+            BatteryType::Keyboard => "input-keyboard-symbolic",
+            BatteryType::Pda | BatteryType::Phone => "smartphone-symbolic",
+            BatteryType::MediaPlayer => "multimedia-player-symbolic",
+            BatteryType::Tablet => "tablet-symbolic",
+            BatteryType::Computer => "laptop-symbolic",
+            BatteryType::GamingInput => "input-gaming-symbolic",
+            BatteryType::Pen => "input-tablet-symbolic",
+            BatteryType::Touchpad => "input-touchpad-symbolic",
+            BatteryType::Network => "network-wired-symbolic",
+            BatteryType::Headset => "audio-headset-symbolic",
+            BatteryType::Speakers => "speaker-symbolic",
+            BatteryType::Headphones => "audio-headphones-symbolic",
+            BatteryType::Video => "video-display-symbolic",
+            BatteryType::OtherAudio => "audio-speakers-symbolic",
+            BatteryType::Printer => "printer-network-symbolic",
+            BatteryType::Scanner => "scanner-symbolic",
+            BatteryType::Camera => "camera-photo-symbolic",
+            _ => "bluetooth-symbolic",
```

Comment:
A few devices don't have an [icon](https://www.figma.com/design/yFxlLCtVYLHMuHBNSXLf7O/Icon-theme?node-id=514-3593&node-type=SYMBOL&t=5S42FVRN9kxzClNA-0). Should I create an issue?